nixos/modules/services/hardware/keyd.nix

```diff
@@ -35,24 +32,64 @@ in
           };
         };
         description = lib.mdDoc ''
-          Configuration, except `ids` section, that is written to {file}`/etc/keyd/default.conf`.
+          Configuration, except `ids` section, that is written to {file}`/etc/keyd/<keyboard>.conf`.
+          Appropriate names can be used to write non-alpha keys, for example "equal" instead of "=" sign (see <https://github.com/NixOS/nixpkgs/issues/236622>).
           See <https://github.com/rvaiya/keyd> how to configure.
         '';
       };
     };
   };
+in
+{
+  options.services.keyd = {
+    enable = mkEnableOption (lib.mdDoc "keyd, a key remapping daemon");
+
+    keyboards = mkOption {
+      type = types.attrsOf (types.submodule keyboardOptions);
+      default = { };
+      example = literalExpression ''
+        {
+          default = {
+            ids = [ "*" ];
+            settings = {
+              main = {
+                capslock = "overload(control, esc)";
+              };
+            };
+          };
+          external_keyboard = {
```

Comment:
This is just an example and this name affects the file name (`external_keyboard.conf`) so I don't think it really matters :)

Reply:
In nix lang `externalKeyboard` should be used.  However, as a file name, `external_keyboard` may be preferred.  I prefer `externalKeyboard` because I consider the generated file as an implementation detail.

That's why I nitpick.

Reply:
Oh, I see what you mean. I'll fix that.